packages/installer/upgrade.sh

```diff
@@ -58,17 +58,22 @@ find_upgrade_channel() {
         source /etc/cos-upgrade-image
     fi
 
+    if [ -n "$NO_CHANNEL" ] && [ $NO_CHANNEL == true ]; then
+        CHANNEL_UPGRADES=false
+    fi
+
     if [ -n "$IMAGE" ]; then
         UPGRADE_IMAGE=$IMAGE
         echo "Upgrading to image $UPGRADE_IMAGE"
-    fi
+    else
 
-    if [ -z "$UPGRADE_IMAGE" ]; then
-        UPGRADE_IMAGE="system/cos"
-    fi
+        if [ -z "$UPGRADE_IMAGE" ]; then
+            UPGRADE_IMAGE="system/cos"
+        fi
 
-    if [ -n "$UPGRADE_RECOVERY" ] && [ $UPGRADE_RECOVERY == true ] && [ -n "$RECOVERY_IMAGE" ]; then
-        UPGRADE_IMAGE=$RECOVERY_IMAGE
+        if [ -n "$UPGRADE_RECOVERY" ] && [ $UPGRADE_RECOVERY == true ] && [ -n "$RECOVERY_IMAGE" ]; then
```

Comment:
UPGRADE_RECOVERY was not set here since it gets applied in from the command line parsing which used to happen after this method in former implementation. Hence the recovery image from `/etc/cos-upgrade-image` was never applied and it was defaulting to  `system/cos`